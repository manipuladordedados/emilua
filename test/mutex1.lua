-- Given Emilua's lock() uses dispatch semantics, a separate test just to ensure
-- check-suspend still happens make sense

local mutex = require('mutex')

m = mutex.new()
this_fiber.forbid_suspend()
m:lock()
