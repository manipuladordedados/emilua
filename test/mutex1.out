Main fiber from VM 0x1 panicked: 'EPERM within a forbid-suspend block'
stack traceback:
	input.lua:8: in main chunk
	[C]: in function ''
	[string "?"]: in function <[string "?"]:0>
